ASAA-31-Repo Branches Screen

LuisRocketRangel opened the pull request:
Requirements:
-Added a Branches item to the Home Nav Drawer
-Added a Branch Screen to display a Repo's branches -Use a UIModel Item for Branches
-Added a Search Bar View

Solution:
-Did above

![Screenshot_20221014-134246](https://user-images.githubusercontent.com/83731304/195921540-45f832e5-25ea-4581-aa1e-c2f3792f6777.png)
![Screenshot_20221014-134250](https://user-images.githubusercontent.com/83731304/195921541-9571e1ab-04c0-49f4-8cc4-0bb85dcd116f.png)

LuisRocketRangel committed:
ASAA-31-Repo Branches Screen

Requirements:
-Added a Branches item to the Home Nav Drawer
-Added a Branch Screen to display a Repo's branches
-Use a UIModel Item for Branches
-Added a Search Bar View

Solution:
-Did above

IvanTelisman commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryBranchesViewModel.kt:
```diff
@@ -0,0 +1,72 @@
+    private val clock by inject<Clock>()
+
+    // State
+    private val srcBranches = MutableStateFlow<List<BranchDto>>(emptyList())
```
Just had a quick chat with Luis about taking these DTO objects out of the VM

LuisRocketRangel replied:
Made the change, DTO is donzo

LuisRocketRangel committed:
ASAA-31-PR Feedback(Ivan)

Requirements:
-Remove BranchDTO from VM

Solution:
-Changed the BranchDTO to the Branch model

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryBranchesViewModel.kt:
```diff
@@ -0,0 +1,68 @@
+                RepositoryBranchItemUiModel(
+                    name = branch.name,
+                    timeSinceCreated = branch.date.formattedUpdateTime(clock),
+                    status = "#1(OPEN)"
```
@LuisRocketRangel Can we derive this data from something else? Is this data overwritten at some point?

* If we need to keep it and it is actually shown in the UI at some point, let's move it to `strings.xml` and probably change `RepositoryBranchItemUiModel.status` from a `String` to `StringIdHelper`
* If not, replace this with the empty string.

LuisRocketRangel replied:
The status is dynamic but the branch call doesn't contain this data. Replaced with the StringIDHelper

IvanTelisman commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBranchesScreen.kt:
```diff
@@ -0,0 +1,179 @@
+            .padding(Dimens.grid_0_25)
+            .border(2.dp, brown_grey, shape = RoundedCornerShape(4.dp))
+    ) {
+        TextField(
```
Hey @LuisRocketRangel, looks like we'll also want to add an attribute to only have this allow 1 line or something like that because I noticed that you can tap enter on the keyboard and it takes you to the next line and causes some odd behavior with the search functionality.

LuisRocketRangel replied:
@IvanTelisman Hey I just added some changes that should take care of this

IvanTelisman replied:
Looks good!

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryBranchesViewModel.kt:
```diff
@@ -0,0 +1,68 @@
+        path.setValue(name)
+        selectedRepo?.let {
+            val slug = it.workspace?.slug ?: ""
+            Timber.d(slug + " " + selectedRepo.name.orEmpty())
```
@LuisRocketRangel nit: Remove this log OR modify it to include the function name as below:
```suggestion
            Timber.d("[getRepoBranches] $slug ${selectedRepo.name.orEmpty()})
```

LuisRocketRangel replied:
Removed log

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/repositoryBranchesComposable.kt:
```diff
@@ -0,0 +1,27 @@
+package com.bottlerocketstudios.brarchitecture.ui.repository
```
@LuisRocketRangel Refactor the filename to be UpperCamelCased, `RepositoryBranchesComposable.kt`. Leave the extension function as is :)

LuisRocketRangel replied:
Fixed!

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBranchesScreen.kt:
```diff
@@ -0,0 +1,179 @@
+                itemCount = state.itemCount.value
+            )
+        }
+        item {
+            SearchView(state = searchState, placeHolder = stringResource(id = R.string.home_branches_search))
+        }
+        items(items = state.branchItems.value.filter { it.name.contains(searchState.value.text) }) {
+            BranchItem(
+                item = it
+            )
+        }
```
@LuisRocketRangel Thanks for adding the Search view! This is a nice addition :)

LuisRocketRangel replied:
Woo!

jaredstockton reviewed (changes requested):
@LuisRocketRangel Great job! I left a few comments for small things to address. Once they've been address, I'll take another look prior to approval :)

LuisRocketRangel committed:
ASAA-31-PR Feedback(Jared)

Requirements:
-Refactor Filenames for camelcase
-Handle the status string better
-Clean up a log

Solution:
-Did above

jaredstockton reviewed (approved):
All changes look good. Thanks for the help!

LuisRocketRangel committed:
ASAA-31-PR Feedback(Search View)

Requirements:
-Dismiss Keyboard when return is hit
-Prevent Newlines

Solution:
-Did above

IvanTelisman commented:
Looks good to me, nice work @LuisRocketRangel !
